File: Stack/1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        Set<Character> vowels = Set.of('a', 'e', 'i', 'o', 'u');
        int l = 0;
        int maxVowels = 0;
        int currVowels = 0;

        for(int r=0;r<s.length();r++) {
            if (vowels.contains(s.charAt(r))) {
                currVowels++;
            }

            if (r-l+1 > k) {
                if(vowels.contains(s.charAt(l))) {
                    currVowels--;
                }
                l++;
            }

            if(r-l+1 == k) {
                maxVowels = Math.max(maxVowels, currVowels);
            }
        }
        return maxVowels;


    }
}

File: Array/1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        count  = 0

        vowels = set({'a', 'e', 'i', 'o', 'u'})
        max_vowels = 0
        l = 0
        for r in range(len(s)):
            if s[r] in vowels:
                count+=1
            while l < len(s) and r-l+1 > k:
                if s[l] in vowels:
                    count-=1
                l+=1
            max_vowels = max(count, max_vowels)
        return max_vowels
            

        